corehq/apps/hqadmin/management/commands/force_web_user_password_reset.py

```diff
@@ -0,0 +1,90 @@
+
+    def handle(self, web_users_file, skip_if_reset_since, **options):
+        web_user_usernames = get_lines_from_file(web_users_file)
+        errors = []
+        web_users_to_reset = []
+        web_users_to_skip_due_to_recent_reset = []
+        for web_user_username in web_user_usernames:
+            web_user = WebUser.get_by_username(web_user_username)
+            if web_user is None:
+                errors.append(f'No user named {web_user_username}')
+            elif not web_user.is_web_user():
+                errors.append(f'{web_user_username} is not a WebUser')
+            elif skip_if_reset_since and web_user.last_password_set \
+                    and web_user.last_password_set > skip_if_reset_since:
+                web_users_to_skip_due_to_recent_reset.append(web_user)
+            else:
+                web_users_to_reset.append(web_user)
+
+        if errors:
+            raise CommandError(f'The following errors were found in your input file:\n{errors}')
+
+        if web_users_to_skip_due_to_recent_reset:
+            self.stdout.write("The following users will be skipped "
+                              "because they have reset their passwords since {skip_if_reset_since}")
+
+            for web_user in web_users_to_skip_due_to_recent_reset:
+                self.stdout.write(f"  - {web_user.username} (password last set on {web_user.last_password_set})")
+
+        if not web_users_to_reset:
+            print("No web users to reset passwords for. No action taken.")
+            return
+
+        self.stdout.write("The following users will be logged out and have their passwords force reset.")
+        self.stdout.write("Additionally, they will receive an email directing them to reset their password.")
+        for web_user in web_users_to_reset:
+            self.stdout.write(f"  - {web_user.username}")
+
+        if 'y' != input('Do you want to proceed? [y/N]'):
+            raise CommandError('You have aborted the command and no action was taken.')
+
+        for web_user in web_users_to_reset:
+            self.stdout.write(f"Force resetting password for {web_user.username}", ending='')
+            force_password_reset(web_user)
+            self.stdout.write(" - Done")
+
+
+def get_lines_from_file(filename):
+    with open(filename) as f:
+        return [line.strip() for line in f.readlines()]
+
+
+def force_password_reset(web_user):
+    user = web_user.get_django_user()
+    user.set_password(uuid.uuid4().hex)
+    user.save()
+    url = f"{get_url_base()}{reverse('password_reset_email')}"
+    send_html_email_async.delay(
+        'Reset Password on CommCare HQ',
+        web_user.get_email(),
+        (f'Your system administrator has forced a password reset. '
+         f'Before you will be able to continue to use CommCare, '
+         f'you must reset your password by navigating to {url} and following the instructions.'),
+        email_from=settings.DEFAULT_FROM_EMAIL
+    )
```

Comment:
Do we need to think about the scenario (assuming it exists) where the user is not linked to a valid email address? `get_email()` returns the email, if set, or the username. Neither of those are guaranteed to be a valid email. Presumably, a valid email was established initially, to verify the account, but after account creation, that can change. If we fail at delivering a password reset notification, are we concerned that we'll be locking out these users with no notification, or can we just expect them to contact their managers?

Reply:
Users are expected to maintain a working email (remember these are people doing this professionally for their job), so if their email is not working then locking them out forever is actually a very sensical thing to do in this case where the program managers are concerned the credentials may have been compromised. And like you said their recourse would be to contact their managers who would sort it out with support.